src/vs/platform/terminal/common/terminal.ts

```diff
@@ -116,7 +117,7 @@ export interface IPtyHostAttachTarget {
 	workspaceId: string;
 	workspaceName: string;
 	isOrphan: boolean;
-	icon: string | undefined;
+	icon: URI | { light: URI; dark: URI } | { id: string, color?: { id: string } } | undefined;
```

Comment:
> decided against this for the reason you mentioned yesterday - I like being able to see on hover the types I can set iconPath to

This unified type will help simplify the code/api quite a lot, there's so much repetition and the types are all a little different as well (different order, `ThemeIcon` vs `{ id: string, color?: { id: string } }`).

```ts
type TerminalIcon = ThemeIcon | URI | { light: URI; dark: URI };
```

The main reason I'm for keeping them unwrapped in the extension API is to make the API easier to work with for new people. Here it's basically just us and a handful of contributors, and we know what a `TerminalIcon` is so we don't need those ergonomics.